### DISCUSSION TITLE
How to reactively change the value and properties of controls?

### DISCUSSION THREAD

**Discussion by Ivan-1F** [pmndrs/leva#338] [Q&A] [answered]
I have a component using Leva with a state. The state is used in the `useControl` schema:

```typescript
const [step, setStep] = useState(10);

useControls({
  x: {
    value: 0,
    step: step
  }
});
```

However when state is updated, the step of the control is not updated

Demo: https://codesandbox.io/s/leva-dynamically-change-properties-oi411h?file=/src/App.tsx

Expected: the step of `x` should be the same as the value in the `<input />`
Actually Happened: the step is always the default value `10`

**Comment by dbismut** [comment, marked as answer]
Put `step` in a dependency array as the last argument of `useControls`.

**Reply by Ivan-1F** [reply]
Thanks a lot! It helps!
